portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -45,11 +41,21 @@ public void init() {
   }
   
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query commentKeysQuery = new Query("Comment").setKeysOnly();
-    PreparedQuery commentKeysResults =  datastore.prepare(commentKeysQuery);
-    for(Entity commentKey : commentKeysResults.asIterable()){
-      datastore.delete(commentKey.getKey());
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Key commentKey = KeyFactory.createKey("Comment", Long.parseLong(request.getParameter("id")));
+    try {
+      Entity comment = datastore.get(commentKey);
+      
+      // Verify the deleter is the author of the comment being deleted. If not, early return.
+      UserService userService = UserServiceFactory.getUserService();
+      if (!userService.isUserLoggedIn()
+          || !comment.getProperty("email").equals(userService.getCurrentUser().getEmail())) {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+        return;
+      }
+      datastore.delete(commentKey);
+    } catch (com.google.appengine.api.datastore.EntityNotFoundException e) {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
You should return after this because sending both the error and the redirect does not make much sense (they have different status codes).